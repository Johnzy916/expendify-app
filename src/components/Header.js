import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom'
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout, userName }) => (
  <header className="header">
      <div className="content-container">
        <nav className="header__content">
          <Link
            className="header__title"
            to="/dashboard">
              <h1>Expendify</h1>
          </Link>
          <div>
            <span className="header__username">
              {userName}
            </span>
            <button
              className="btn btn--primary btn--link"
              onClick={startLogout}>
                Logout
            </button>
          </div>
        </nav>
      </div>
  </header>
);

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout())
});

const mapStateToProps = state => ({
  userName: state.auth.userName
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);