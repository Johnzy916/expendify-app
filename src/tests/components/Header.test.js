import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';
import '../setupTests.js'; // needed for codesandbox.io
import toJSON from 'enzyme-to-json'; // needed for codesandbox.io

test('should render Header correctly', () => {
  const wrapper = shallow(<Header />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});