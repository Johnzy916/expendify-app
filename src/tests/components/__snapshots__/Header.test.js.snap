// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <nav
      className="header__content"
    >
      <Link
        className="header__title"
        to="/dashboard"
      >
        <h1>
          Expendify
        </h1>
      </Link>
      <div>
        <span
          className="header__username"
        />
        <button
          className="btn btn--primary btn--link"
          onClick={[Function]}
        >
          Logout
        </button>
      </div>
    </nav>
  </div>
</header>
`;
